clients/privacy-center/components/consent/NoticeDrivenConsent.tsx

```diff
@@ -39,15 +42,33 @@ import SaveCancel from "./SaveCancel";
 import PrivacyPolicyLink from "./PrivacyPolicyLink";
 
 // DEFER(fides#3505): Use the fides-js version of this function
-const resolveConsentValue = (
+export const resolveConsentValue = (
   notice: PrivacyNoticeResponseWithUserPreferences,
-  context: ConsentContext
-) => {
+  context: ConsentContext,
+  cookie: FidesCookie
+): UserConsentPreference | undefined => {
+  if (notice.consent_mechanism === ConsentMechanism.NOTICE_ONLY) {
+    return UserConsentPreference.ACKNOWLEDGE;
+  }
   const gpcEnabled =
-    !!notice.has_gpc_flag && context.globalPrivacyControl === true;
+    !!notice.has_gpc_flag &&
+    context.globalPrivacyControl === true &&
+    !noticeHasConsentInCookie(notice as PrivacyNoticeWithPreference, cookie);
   if (gpcEnabled) {
     return UserConsentPreference.OPT_OUT;
   }
+  const preferenceExistsInCookie = noticeHasConsentInCookie(
+    notice as PrivacyNoticeWithPreference,
+    cookie
+  );
+  if (preferenceExistsInCookie) {
+    return transformConsentToFidesUserPreference(
+      // @ts-ignore
+      cookie.consent[notice.notice_key],
```

Comment:
is the ts problem that this could be undefined? if so, doing a `!!` would be safer than a `@ts-ignore`
```suggestion
      !!cookie.consent[notice.notice_key],
```

Reply:
This is the ts issue:

<img width="687" alt="Screenshot 2024-01-04 at 5 41 23 PM" src="https://github.com/ethyca/fides/assets/7697292/cd027e5e-34a4-4572-b780-23e67341d0cb">